setup.py

```diff
@@ -58,7 +58,7 @@
 extras_require = {
     "gpu": [
         "nvidia-ml-py3>=7.352.0",
-        "tensorflow==2.8.4",  # FIXME: Temporarily pinned due to issue with TF version > 2.10.1 See #2018
+        "tensorflow>=2.8.4,!=2.9.0.*,!=2.9.1,!=2.9.2,!=2.10.0.*,<2.12",
```

Comment:
```suggestion
        "tensorflow>=2.8.4,!=2.9.0.*,!=2.9.1,!=2.9.2,!=2.10.0.*,<3",
```

Can we try removing `<2.12`, and maybe add <3? otherwise we will be constraining TF to old versions